pkg/webui/oauth/views/change-password/index.js

```diff
@@ -0,0 +1,170 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { withRouter } from 'react-router-dom'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+import { replace } from 'connected-react-router'
+import { connect } from 'react-redux'
+import * as Yup from 'yup'
+
+import api from '../../api'
+import sharedMessages from '../../../lib/shared-messages'
+
+import Button from '../../../components/button'
+import Form from '../../../components/form'
+import Input from '../../../components/input'
+import SubmitButton from '../../../components/submit-button'
+import Logo from '../../../components/logo'
+import IntlHelmet from '../../../lib/components/intl-helmet'
+import Message from '../../../lib/components/message'
+
+import style from '../login/login.styl'
+
+const m = defineMessages({
+  changePassword: 'Change Password',
+  inputNewPassword: 'Input a new password',
+  passwordChanged: 'Password successfully changed',
+  loginPage: 'Go to Login page'
+})
+
+const digit = /(?=.*[\d])/
+const uppercase = /(?=.*[A-Z])/
+const special = /(?=.*[!@#$%^&*])/
```

Comment:
move those to the shared file `oauth/lib/regexp.js`; the register page has the same definitions